### PR TITLE
ci fix

### PR DESCRIPTION
**Pull request by LetterN**
## About The Pull Request
attempt to fix ci using https://github.com/discordia-space/CEV-Eris/pull/7406 . the ci clears z1 as its used for spawning stuff

## Changelog
:cl:
code: ci fix.
code: update apc initialize/new code
code: machine checking is during lateinit for areas
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
No way

**Comment by MLGTASTICa** [comment]
Tests aren't functional , Science genetics have no vent/scrubber , and theres /area/engineering/break_room (not breakroom) that only has 1 tile and lacks both an APC and any vent and scrubbers.

File: code/__HELPERS/_logging.dm [hunk 1]
```diff
@@ -116,6 +116,16 @@
...
 
+/// Logging for mapping errors
+/proc/log_mapping(text, skip_world_log)
+#ifdef UNIT_TESTS
+	GLOB.unit_test_mapping_logs += text
+#endif
+	log_world("## MAPPING: [text]")
+	if(skip_world_log)
+		return
+	SEND_TEXT(world.log, text)
```
Comment (hyperioo) [inline code/__HELPERS/_logging.dm:127]:
```suggestion
	log_world(text)
```
Seems there is already a function for that

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -163,55 +165,96 @@
...
+	if(cell_type)
+		cell = new cell_type(src)
+		cell.charge = start_charge * cell.maxcharge / 100 // (convert percentage to actual value)
+
+	var/area/our_area = get_area(loc)
+
+	//if area isn't specified use current
+	if(areastring)
+		area = get_area_name(areastring)
+		if(!area)
```
Comment (hyperioo) [inline code/modules/power/apc.dm:217]:
```suggestion
		if(!area && isarea(our_area))
```

File: code/modules/unit_tests/unit_test.dm [hunk 3]
```diff
@@ -11,6 +11,8 @@ You can use the run_loc_floor_bottom_left and run_loc_floor_top_right to get tur
 
 */
 
+/// IMPORTANT !!! IF YOU ARE CHANGING CC CHANGE THIS TOO
```
Comment (hyperioo) [inline code/modules/unit_tests/unit_test.dm:14]: CC ?

**Review by hyperioo — APPROVED** [review]

**Pull request merged**